projects/storefrontlib/src/cms-components/myaccount/order/order-details/order-detail-items/order-detail-items.component.ts

```diff
@@ -16,7 +16,10 @@ export class OrderDetailItemsComponent implements OnInit {
   constructor(protected orderDetailsService: OrderDetailsService) {}
 
   promotionLocation: PromotionLocation = PromotionLocation.Order;
-  order$: Observable<any> = this.orderDetailsService.getOrderDetails();
+
+  order$: Observable<Order> = this.orderDetailsService
+    .getOrderDetails()
+    .pipe(map((order) => (Object.keys(order).length ? order : null)));
```

Comment:
Could you add unit tests that make sure we don't display anything when we don't have order?